code/obj/item/misc_weapons.dm

```diff
@@ -486,35 +486,41 @@
 		..()
 		BLOCK_SETUP(BLOCK_KNIFE)
 
-/obj/item/dagger/throw_impact(atom/A, datum/thrown_thing/thr)
-	if(iscarbon(A))
-		if (ismob(usr))
-			A:lastattacker = usr
-			A:lastattackertime = world.time
-		A.changeStatus("weakened", 6 SECONDS)
-		A:force_laydown_standup()
-		take_bleeding_damage(A, null, 5, DAMAGE_CUT)
-		playsound(src, 'sound/impact_sounds/Flesh_Stab_3.ogg', 40, 1)
+	overwrite_impact_sfx(original_sound, hit_atom, thr)
+		..()
+		if(ismob(hit_atom))
+			. = 'sound/impact_sounds/Flesh_Stab_3.ogg'
 
-/obj/item/dagger/attack(target, mob/user)
-	playsound(target, "sound/impact_sounds/Flesh_Stab_1.ogg", 60, 1)
-	if(iscarbon(target))
-		var/mob/living/carbon/C = target
-		if(!isdead(C))
-			take_bleeding_damage(C, user, 5, DAMAGE_STAB)
-	..()
+
+	throw_impact(atom/A, datum/thrown_thing/thr)
+		if (..())
+			return
+		if(ismob(A))
+			var/mob/M = A
+			if (ismob(usr))
+				M:lastattacker = usr
+				M:lastattackertime = world.time
+			M.changeStatus("weakened", 6 SECONDS)
+			M.force_laydown_standup()
+			take_bleeding_damage(M, null, 5, DAMAGE_CUT)
+
+	attack(target, mob/user)
+		playsound(target,  'sound/impact_sounds/Flesh_Stab_1.ogg ', 60, 1)
+		if(ismob(target))
+			take_bleeding_damage(target, user, 5, DAMAGE_STAB)
+		..()
 
 /obj/item/dagger/smile
 	name = "switchblade"
 	force = 10.0
 	throw_range = 10
 	throwforce = 10.0
 
-/obj/item/dagger/smile/attack(mob/living/target, mob/user)
-	if(prob(10))
-		var/say = pick("Why won't you smile?","Smile!","Why aren't you smiling?","Why is nobody smiling?","Smile like you mean it!","That is not a smile!","Smile, [target.name]!","I will make you smile, [target.name].","[target.name] didn't smile!")
-		user.say(say)
-	..()
+	attack(mob/living/target, mob/user)
```

Comment:
Absolute proc paths are generally preferred I think, better to leave it like that.

Reply:
every day a proc is made relative, an angel loses its wings

Reply:
*hurriedly hides several hundred lines of relative proc commits*